frontend/server/src/Controllers/Session.php

```diff
@@ -520,6 +533,62 @@ public static function nativeLogin(\OmegaUp\Request $r): string {
         }
     }
 
+    /**
+     * Does login for an identity given username or email. Password no needed
+     * because user should have a successful native login
+     *
+     * @omegaup-request-param null|string $auth_token
+     */
+    public static function loginWithAssociatedIdentity(
+        \OmegaUp\Request $r,
+        string $usernameOrEmail,
+        \OmegaUp\DAO\VO\Identities $loggedIdentity
+    ): void {
+        $response = \OmegaUp\DAO\Identities::resolveAssociatedIdentity(
+            $usernameOrEmail,
+            $loggedIdentity
+        );
+        if (is_null($response)) {
+            self::$log->warn("Identity {$usernameOrEmail} not found.");
+            throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
+        }
+
+        // Only users acting as main identities can select another identity
+        if (!$response['loggedAsMainIdentity']) {
```

Comment:
entonces si un usuario hace login con su identidad principal, cambia a otra identidad ya no podría regresar a la identidad principal? creía que soportar ese caso era el punto de hacer el cambio a la tabla de auth tokens :(

se puede agregar esa prueba? y también otra que haciendo login originalmente con una identidad no-principal, esto no te permita cambiar a otra identidad.

Reply:
Justo es lo que hace, quizás los nombres de las variables o el comentario de arriba no expresan lo que realmente hace esto. 

`loggedAsMainIdentity` es la combinación de campos `user_id` y `acting_identity_id`  en la tabla de `Auth_Tokens`: Si los dos se llenaron quiere decir que el usuario se logueó con una identidad secundaria debido a que está asociada a un `user_id`.

Así se mostraría si las identidades no principales pudieran ver el listado de identidades asociadas:
![LogAsMainIdentity](https://user-images.githubusercontent.com/3230352/98576805-ac7d8e80-2280-11eb-88c5-fb42952f3dca.gif)

Así es como debería quedar, donde sólo las identidades principales pueden ver ese listado:
![LogAsMainIdentityFinal](https://user-images.githubusercontent.com/3230352/98576932-da62d300-2280-11eb-95ca-ccd315235907.gif)


Agrego la prueba para mostrar que si funcione como comentas.

Reply:
a la prueba le faltaba un `$this>fail()` ^^;; tons no estaba funcionando correctamente. pero fuera de eso, también era cosa de nombres:

```suggestion
        // Only users that originally logged in from their main identities can select another identity.
        if (is_null($r->user) || $r->user->main_identity_id != $r->loginIdentity->identity_id.) {
```

Reply:
Listo